### PR TITLE
feat(secrets): enable rule and report `warn`

### PR DESCRIPTION
**Pull request by philippfromme**
Related to https://github.com/camunda/camunda-modeler/issues/3834

### CODE REVIEW

**Comment by nikku** [comment]
Did we settle on _info_ over _warning_? To date _info_ is reserved for modeling only things.

**Comment by philippfromme** [comment]
> Did we settle on _info_ over _warning_? To date _info_ is reserved for modeling only things.

Sorry, I misread your [comment](https://github.com/camunda/camunda-modeler/issues/3834#issuecomment-1713540141) will change to `warn`.

**Comment by philippfromme** [comment]
@nikku Reported as `warn` now. Interaction with the properties panel will still be the same. No error but you can still focus the properties panel input.

File: test/config/configs.spec.js [hunk 1]
```diff
@@ -135,6 +135,7 @@ describe('configs', function() {
     'no-propagate-all-parent-variables' : [ 'error', { version: '8.0' } ],
     'no-task-schedule': [ 'error', { version: '8.0' } ],
     'no-zeebe-properties': [ 'error', { version: '8.0' } ],
+    'secrets': [ 'warn', { version: '8.0' } ],
```
Comment (nikku) [inline test/config/configs.spec.js:138]: From what I understand this will change for `8.3` and up only. Let's clarify this.

Comment (philippfromme) [reply test/config/configs.spec.js:138]: This is based on https://github.com/camunda/bpmnlint-plugin-camunda-compat/pull/116#issuecomment-1678576712. Should we ask them again?

Comment (nikku) [reply test/config/configs.spec.js:138]: 8.0 as an answer makes no sense to me, as 8.0 ships with an entirely different connector run-time (that does not know the new pattern yet).

**Review by nikku — APPROVED** [review]

**Pull request merged**